Dangerfile.js

```diff
@@ -0,0 +1,46 @@
+// Dangerfile.js
+import { danger } from 'danger';
+import fs from 'fs';
+import path from 'path';
+
+// Define couplings in a way that each key appears only once
+const couplings = {
+  '.github/workflows/check-file-diff.yml': [
+    'package.json',
+    'babel.config.js',
+  ],
+  'babel.config.js': [
+    'package.json',
+    'lerna.json',
+  ],
+};
+
+const { git } = danger;
+
+Promise.resolve().then(() => {
+  const allModifiedFiles = [...git.modified_files, ...git.created_files];
+  const messages = [];
+
+  Object.entries(couplings).forEach(([implFile, testOrDocFiles]) => {
+    testOrDocFiles.forEach((testOrDocFile) => {
+      if (allModifiedFiles.includes(implFile) && !allModifiedFiles.includes(testOrDocFile)) {
+        messages.push(`Changes to ${implFile} should include changes to ${testOrDocFile}`);
+      }
+    });
+  });
+
+  const filePath = path.join(process.env.GITHUB_ENV || '.', 'action_output');
+  if (messages.length > 0) {
+    const messageString = messages.join('\n');
+    fs.appendFileSync(filePath, `RESULT<<EOF\n${messageString}\nEOF\n`);
```

Comment:
hmmm should this be appending? or should a new file be created at the start? Will the old error logs also be included?